src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4390,10 +4390,13 @@ public String descriptorString() {
      *
      * Returns an array containing {@code Class} objects representing the
      * direct subinterfaces or subclasses permitted to extend or
-     * implement this class or interface if it is sealed. The order of such elements
-     * is unspecified. If this {@code Class} object represents a primitive type,
+     * implement this class or interface if it is sealed.  The order of such elements
+     * is unspecified. The array is empty if this sealed class or interface has no
+     * permitted subclass. If this {@code Class} object represents a primitive type,
      * {@code void}, an array type, or a class or interface that is not sealed,
-     * an empty array is returned.
+     * that is {@link #isSealed()} returns {@code false}, then this method returns {@code null}.
+     * Conversely, if {@link #isSealed()} returns {@code true}, then this method
+     * returns a non-null value.
```

Comment:
@ChrisHegarty a simpler alternative to address your concern is to add a link to the reference to "sealed class or interface" to `Class::isSealed` as follows:
```
implement this class or interface if it is {@linkplain #isSealed() sealed}.
The order of such elements is unspecified. If this {@code Class} object
represents a primitive type,  is unspecified. The array is empty if this
{@linkplain #isSealed() sealed} class or interface has no permitted subclass. 
If this {@code Class} object represents a primitive type, {@code void}, an array type,
or a class or interface that is not sealed, then this method returns {@code null}.
```

Reply:
There is certainly a little redundancy in the additional spec wording
that I proposed. In my view it is worth it, as it allows the reader to
more easily grok the null versus empty array for non-sealed classes,
without having to navigate between the pair of methods.

Reply:
I see this new pattern introduced in `getRecordComponents`.  You may consider if this pattern should consistently applied in other `Class` APIs such as `getEnumConstants`.